### DISCUSSION TITLE
TypeScript, Framer motion and Chakra UI

### DISCUSSION THREAD

**Discussion by niels-k-86** [chakra-ui/chakra-ui#5213] [Q&A]
I'm trying to use Framer Motion in my Chakra UI project. I'm using TypeScript. I've read through the Chakra documentation regarding the use of the `as` prop, and how to wrap Chakra components in the Framer Motion `motion()` function. I want to combine these two things but I'm running into problems. Here's what I'm trying to do:

```
// MotionBox.tsx

import * as React from "react";
import { Box, BoxProps } from "@chakra-ui/react";
import { CustomDomComponent, motion } from "framer-motion";

export type MotionBoxProps = CustomDomComponent<BoxProps>;

const MotionBox = motion<BoxProps>(Box);

export default MotionBox;

```

So far so good. I can use this `MotionBox` element including it's Framer Motion props (`animate`, `layout`, etc.) and still use the `as` prop from Chakra.

However, things get complicated when trying to use this `MotionBox` as the base for another component. Here's an example of what I'm trying to do:

```
// MyCustomComponent.tsx

import * as React from "react";
import { BoxProps, forwardRef } from "@chakra-ui/react";
import MotionBox, { MotionBoxProps } from "./motion/MotionBox";

export type MyCustomComponentProps = MotionBoxProps & {
  aCustomProp?: string;
};

const MyCustomComponent = forwardRef<MyCustomComponentProps, "div">(
  ({ children, aCustomProp, ...restProps }, ref) => {
    return (
      <MotionBox ref={ref} {...restProps}>
        {children}
      </MotionBox>
    );
  }
);

export default MyCustomComponent;
```
The error I'm getting:

<img width="1599" alt="Schermafbeelding 2021-12-09 om 16 09 09" src="https://user-images.githubusercontent.com/1876566/145424150-2fefe293-3259-404f-9ba2-1a6c7ff999db.png">

So I tried removing the `forwardRef`, since I doubt I would even need that.

```
// MyCustomComponent.tsx

import * as React from "react";
import { BoxProps, forwardRef } from "@chakra-ui/react";
import MotionBox, { MotionBoxProps } from "./motion/MotionBox";

export type MyCustomComponentProps = MotionBoxProps & {
  aCustomProp?: string;
};

const MyCustomComponent = ({
  children,
  aCustomProp,
  ...restProps
}: MyCustomComponentProps) => {
  return <MotionBox {...restProps}>{children}</MotionBox>;
};

export default MyCustomComponent;
```

I get a new error about `children` not being a prop:

```
Property 'children' does not exist on type 'ForwardRefExoticComponent<BoxProps & MotionProps & RefAttributes<SVGElement | HTMLElement>> & { ...; }'.
```

I can add the `children` prop back through `React.PropsWithChildren` but it's weird that I have to explicitly add this. Am I missing something here?

**Comment by ELI7VH** [comment]
3 yearw with no answer. any luck?
